libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -762,62 +751,54 @@ static IotMqttError_t _subscriptionCommon( IotMqttOperationType_t operation,
                                               pCallbackInfo,
                                               &pSubscriptionOperation );
 
-    if( status != IOT_MQTT_SUCCESS )
-    {
-        goto cleanup;
-    }
-
-    /* Add the subscription list for a SUBSCRIBE. */
-    if( operation == IOT_MQTT_SUBSCRIBE )
+    if( status == IOT_MQTT_SUCCESS )
     {
-        status = _IotMqtt_AddSubscriptions( mqttConnection,
-                                            pSubscriptionOperation->u.operation.packetIdentifier,
-                                            pSubscriptionList,
-                                            subscriptionCount );
-
-        if( status != IOT_MQTT_SUCCESS )
+        /* Add the subscription list for a SUBSCRIBE. */
+        if( operation == IOT_MQTT_SUBSCRIBE )
         {
-            goto cleanup;
+            status = _IotMqtt_AddSubscriptions( mqttConnection,
+                                                pSubscriptionOperation->u.operation.packetIdentifier,
+                                                pSubscriptionList,
+                                                subscriptionCount );
         }
     }
 
-    /* Set the reference, if provided. */
-    _setOperationReference( pOperationReference, pSubscriptionOperation );
-
-    /* Send the SUBSCRIBE packet. */
-    if( ( flags & MQTT_INTERNAL_FLAG_BLOCK_ON_SEND ) == MQTT_INTERNAL_FLAG_BLOCK_ON_SEND )
-    {
-        _IotMqtt_ProcessSend( IOT_SYSTEM_TASKPOOL, pSubscriptionOperation->job, pSubscriptionOperation );
-    }
-    else
+    if( status == IOT_MQTT_SUCCESS )
     {
-        status = _IotMqtt_ScheduleOperation( pSubscriptionOperation,
-                                             _IotMqtt_ProcessSend,
-                                             0 );
+        /* Set the reference, if provided. */
+        _setOperationReference( pOperationReference, pSubscriptionOperation );
 
-        if( status != IOT_MQTT_SUCCESS )
+        /* Send the SUBSCRIBE packet. */
+        if( ( flags & MQTT_INTERNAL_FLAG_BLOCK_ON_SEND ) == MQTT_INTERNAL_FLAG_BLOCK_ON_SEND )
+        {
+            _IotMqtt_ProcessSend( IOT_SYSTEM_TASKPOOL, pSubscriptionOperation->job, pSubscriptionOperation );
+        }
+        else
         {
-            IotLogError( "(MQTT connection %p) Failed to schedule %s for sending.",
-                         mqttConnection,
-                         IotMqtt_OperationType( operation ) );
+            status = _IotMqtt_ScheduleOperation( pSubscriptionOperation,
+                                                 _IotMqtt_ProcessSend,
+                                                 0 );
 
-            if( operation == IOT_MQTT_SUBSCRIBE )
+            if( status != IOT_MQTT_SUCCESS )
```

Comment:
The same thing as line 1588 can be done here by moving the assignment and comparison into an `else if` to reduce a nesting level. Not necessary for this PR though. In both cases, we'd need an empty `else` to comply with MISRA [15.7](https://rules.sonarsource.com/c/RSPEC-126)

Reply:
I avoided empty else statements ,  unless it is  necessary to  reduce complexity we should not use it. Since this function has higher complexity, I'll try to make the change.